src/Microsoft.Azure.WebJobs.Host.Storage/IAzureStorageProvider.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Azure.Storage.Blobs;
+
+namespace Microsoft.Azure.WebJobs.Host.Storage
+{
+    /// <summary>
+    /// Interface to retrieve BlobServiceClient objects
```

Comment:
Based on this description that the interface is only for serving up BlobServiceClients, the name IAzureStorageProvider seems too general. It doesn't even seem like this interface even needs to exist. Why isn't all this BlobServiceClient functionality just part of BlobServiceClientProvider? It looks like your default AzureStorageProvider implementation is just delegating there anyways.

Reply:
The goal of this interface was to create a layer on top of the individual Azure storage clients (i.e. blobs, queues, files, tables). All these clients would handle configuration similarly so having an object that handles the configuration would be useful. Right now, only Blobs is needed (open for discussion on whether to keep this interface).